File: app/providers/fake.py
from datetime import UTC, datetime

from app.providers.base import TweetData, TweetProvider


class FakeTweetProvider(TweetProvider):
    async def get_tweet(self, tweet_id: str, source_url: str) -> TweetData:
        return TweetData(
            tweet_id=tweet_id,
            url=source_url,
            author_name="Xtract Demo",
            author_username="xtract_demo",
            author_url="https://x.com/xtract_demo",
            text=f"Demo tweet {tweet_id}. Replace TWEET_PROVIDER to fetch real X/Twitter posts.",
            created_at=datetime.now(UTC),
            media=[],
            lang="en",
        )

File: src/providers/fake.ts
import { makeTweet, type TweetData, type TweetProvider } from "@/providers/base";

export class FakeTweetProvider implements TweetProvider {
  async getTweet(tweetId: string, sourceUrl: string): Promise<TweetData> {
    return makeTweet({
      tweetId,
      url: sourceUrl,
      authorName: "Xtract Demo",
      authorUsername: "xtract_demo",
      authorUrl: "https://x.com/xtract_demo",
      authorAvatarUrl: "https://pbs.twimg.com/profile_images/xtract_demo.jpg",
      text: `Demo tweet ${tweetId}. Replace TWEET_PROVIDER to fetch real X/Twitter posts.`,
      createdAt: new Date(),
      lang: "en",
    });
  }

  async health(): Promise<boolean> {
    return true;
  }

  async close(): Promise<void> {
    // no-op
  }
}
